src/main/perf/LineFileDocs.java

```diff
@@ -750,24 +759,28 @@ private static class LineFileDoc {
     final String stringText;
     final ByteBuffer byteText;
 
-    LineFileDoc(String text, float[] vector) {
+    final int idOrBase; // will be the exact id if this is txt based LFD, otherwise will be id base
+
+    LineFileDoc(String text, float[] vector, int id) {
       stringText = text;
       byteText = null;
       if (vector == null) {
         this.vector = null;
       } else {
         this.vector = FloatBuffer.wrap(vector);
       }
+      this.idOrBase = id;
```

Comment:
I think `idOrBase` is really confusing ;)  Is it really sometimes an `id` and sometimes the `base` for the block?

Reply:
LOL this is due to we're sharing a same class for txt LFD and binary LFD, perhaps it's time to give it a better abstraction.